firestore/src/common/firestore.cc

```diff
@@ -309,13 +309,20 @@ WriteBatch Firestore::batch() const {
 
 Future<void> Firestore::RunTransaction(
     std::function<Error(Transaction&, std::string&)> update) {
+  return RunTransaction({}, std::move(update));
+}
+
+Future<void> Firestore::RunTransaction(
+    TransactionOptions options,
+    std::function<Error(Transaction&, std::string&)> update) {
   if (!update) {
     SimpleThrowInvalidArgument(
         "Transaction update callback cannot be an empty function.");
   }
 
   if (!internal_) return FailedFuture<void>();
-  return internal_->RunTransaction(firebase::Move(update));
+  return internal_->RunTransaction(firebase::Move(update),
```

Comment:
`internal_->RunTransaction` should take the option instead of just `max_attempts`. It's OK to choose to do it in the future however, your choice.

Reply:
I had originally coded it with an "internal" version of TransactionOptions actually (e.g. https://github.com/firebase/firebase-cpp-sdk/pull/966/commits/20240d6850dbac8caf2714258dc48ff241542329). However, it ended up just being a lot of boilerplate. I made the call to just take `max_attempts` with the idea that if we ever add another transaction option we can revisit the idea of making an "internal" version of TransactionOptions. Moreover, the Core C++ RunTransaction API (https://github.com/firebase/firebase-ios-sdk/blob/89c38d1ed908dbc10d5f9f7aded4cf271113773f/Firestore/core/src/api/firestore.h#L98-L100) just takes a `max_attempts` instead of a TransactionOptions object. So at some point the C++ SDK has to throw away the TransactionOptions object and just pass max_attempts anyways.